tests/bud.bats

```diff
@@ -1622,13 +1623,18 @@ function _test_http() {
   run_buildah from ${target}
 }
 
-@test "bud-github-context-with-branch-and-subdir" {
+@test "bud-github-context-with-branch,subdir and commit" {
   target=github-image
   gitrepo=https://github.com/containers/podman.git#main:contrib/hello
   run_buildah build $WITH_POLICY_JSON -t ${target} "${gitrepo}"
   # check syntax only for subdirectory
   gitrepo=https://github.com/containers/podman.git#:contrib/hello
   run_buildah build $WITH_POLICY_JSON -t ${target} "${gitrepo}"
+  # Try pulling repo with specific commit
+  # `contrib/helloimage` is only present on or before `40ba9f10e5fbdd3c9d36389107b8bf1caec6cef0`
+  # hence following test verifies if we are fetching the right commit.
+  gitrepo=https://github.com/containers/podman.git#40ba9f10e5fbdd3c9d36389107b8bf1caec6cef0:contrib/helloimage
+  run_buildah build $WITH_POLICY_JSON -t ${target} "${gitrepo}"
```

Comment:
Are we testing the combinatoin of a commit/branch ID but no subdirectory name somewhere else?

Reply:
added a new test for this